src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Formatting/CSharpOnTypeFormattingPass.cs

```diff
@@ -78,8 +77,30 @@ public async override Task<FormattingResult> ExecuteAsync(FormattingContext cont
 
             // We only want to adjust the range that was affected.
             // We need to take into account the lines affected by formatting as well as cleanup.
-            var cleanupLineDelta = LineDelta(formattedText, cleanupChanges);
-            var rangeToAdjust = new Range(rangeAfterFormatting.Start, new Position(rangeAfterFormatting.End.Line + cleanupLineDelta, 0));
+            var lineDelta = LineDelta(formattedText, cleanupChanges);
+
+            // Okay hear me out, I know this looks lazy, but it totally makes sense.
+            // This method is called with edits that the C# formatter wants to make, and from those edits we work out which
+            // other edits to apply etc. Fine, all good so far. BUT its totally possible that the user typed a closing brace
+            // in the same position as the C# formatter thought it should be, on the line _after_ the code that the C# formatter
+            // reformatted.
+            //
+            // For example, given:
+            // if (true){
+            //     }
+            //
+            // If the C# formatter is happy with the placement of that close brace then this method will get two edits:
+            //  * On line 1 to indent the if by 4 spaces
+            //  * On line 1 to add a newline and 4 spaces in front of the opening brace
+            //
+            // We'll happy format lines 1 and 2, and ignore the closing brace altogether. So, by looking one line further
```

Comment:
Super interesting, so is it that we're actually looking one line further or is it that we're looking at the "currently typed on line"? I wonder if we could change the formatter to understand a "line of interest". Asking because I'd worry that looking one line further may then make us look into non-C# land possibly? Or am I being concerned for no reason?